compiler/src/main/java/com/squareup/anvil/compiler/AnvilComponentRegistrar.kt

```diff
@@ -37,25 +27,16 @@ class AnvilComponentRegistrar : ComponentRegistrar {
     val sourceGenFolder = File(configuration.getNotNull(srcGenDirKey))
     val scanner = ClassScanner()
 
-    val codeGenerators = mutableListOf<CodeGenerator>()
+    val generateDaggerFactories = configuration.get(generateDaggerFactoriesKey, false)
     val generateDaggerFactoriesOnly = configuration.get(generateDaggerFactoriesOnlyKey, false)
+    val context = RealAnvilContext(generateDaggerFactories, generateDaggerFactoriesOnly)
+    val codeGenerators = loadCodeGenerators()
+      .filter { it.isApplicable(context) }
+      .toMutableList()
 
     if (!generateDaggerFactoriesOnly) {
-      codeGenerators += ContributesToGenerator()
-      codeGenerators += ContributesBindingGenerator()
-      codeGenerators += ContributesMultibindingGenerator()
+      // TODO do we just special case this one?
```

Comment:
Yes, please remove the TODO and add a comment why.

Reply:
Yep left this for CR discussion 👍. Will update